core/src/kvs/surrealkv/mod.rs

```diff
@@ -63,6 +63,10 @@ impl Datastore {
 	pub(crate) async fn new(path: &str) -> Result<Datastore, Error> {
 		// Create new configuration options
 		let mut opts = Options::new();
+		// Ensure versions are enabled
+		opts.enable_versions = true;
+		// Ensure persistence is enabled
+		opts.disk_persistence = true;
 		// Set the data storage directory
```

Comment:
In case where versions are disabled, there would only be one pair of key-value pair essentially. In this scenario, could we call delete (aka hard_delete) instead of soft_delete because it will help clear unnecessary keys.

Would make sense addressing this post the surrealkv hard delete bug fix release

Reply:
Do you mean this @arriqaaq : https://github.com/surrealdb/surrealdb/pull/5007/commits/319215638064b6c2e95895df677d939991f6a96d ?

Reply:
Yes, if versions are disabled, default to `delete` as it will save memory IMO

Reply:
Yeah I did that just to see if tests would pass (they did) and have reverted it to hard deletes now 👍